src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/DynamicHelpers.cs

```diff
@@ -2,50 +2,90 @@
 
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
+using Microsoft.VisualStudio.Debugger.Evaluation.ClrCompilation;
 using Microsoft.VisualStudio.Debugger.Metadata;
 
 namespace Microsoft.CodeAnalysis.ExpressionEvaluator
 {
-    internal static class DynamicHelpers
+    internal static class AttributeHelpers
     {
-        public static DynamicFlagsCustomTypeInfo GetDynamicFlags(this IList<CustomAttributeData> attributes)
+        internal static DkmClrCustomTypeInfo GetCustomTypeInfo(this IList<CustomAttributeData> attributes)
         {
+            ReadOnlyCollection<byte> dynamicFlags = null;
+            ReadOnlyCollection<string> tupleElementNames = null;
             foreach (var attribute in attributes)
             {
-                if (attribute.Constructor.DeclaringType.IsType("System.Runtime.CompilerServices", "DynamicAttribute"))
+                var attributeType = attribute.Constructor.DeclaringType;
+                if (attributeType.IsType("System.Runtime.CompilerServices", "DynamicAttribute"))
                 {
-                    var arguments = attribute.ConstructorArguments;
-                    if (arguments.Count == 0)
+                    dynamicFlags = GetDynamicFlags(attribute);
+                }
+                else if (attributeType.IsType("System.Runtime.CompilerServices", "TupleElementNamesAttribute"))
+                {
+                    tupleElementNames = GetTupleElementNames(attribute);
+                }
+            }
+            var payload = CustomTypeInfo.Encode(dynamicFlags, tupleElementNames);
+            return (payload == null) ? null : DkmClrCustomTypeInfo.Create(CustomTypeInfo.PayloadTypeId, payload);
```

Comment:
Last two lines could be simplified to `return CustomTypeInfo.Create(dynamicFlags, tupleElementNames);`?